### PR TITLE
id: fix the id allocator is not monotonic

### PR DESCRIPTION
**Pull request by rleungx**
### What is changed and how it works?

This PR tries to make the id monotonic since there are many components that depend on this property. ref: https://github.com/tikv/pd/issues/3303

### Check List

Tests

- Unit test

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### CODE REVIEW

**Review by HunDunDM — APPROVED** [review]

File: server/server.go [hunk 1]
```diff
@@ -1223,6 +1223,10 @@ func (s *Server) campaignLeader() {
 		log.Error("failed to load persistOptions from etcd", errs.ZapError(err))
 		return
 	}
+	if err := s.idAllocator.Generate(); err != nil {
```
Comment (HunDunDM) [inline server/server.go:1226]: I think `Generate` also needs `mu.Lock`?

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**